core/shared/platform/windows/win_file.c

```diff
@@ -1224,18 +1365,96 @@ os_linkat(os_file_handle from_handle, const char *from_path,
           os_file_handle to_handle, const char *to_path,
           __wasi_lookupflags_t lookup_flags)
 {
+#if WINAPI_PARTITION_DESKTOP == 0
+    return __WASI_ENOSYS;
+#else
     CHECK_VALID_FILE_HANDLE(from_handle);
     CHECK_VALID_FILE_HANDLE(to_handle);
 
-    return __WASI_ENOSYS;
+    wchar_t absolute_from_path[PATH_MAX];
+    __wasi_errno_t error = get_absolute_filepath(
+        from_handle->raw.handle, from_path, absolute_from_path, PATH_MAX);
+
+    if (error != __WASI_ESUCCESS)
+        return error;
+
+    wchar_t absolute_to_path[PATH_MAX];
+    error = get_absolute_filepath(to_handle->raw.handle, to_path,
+                                  absolute_to_path, PATH_MAX);
+
+    if (error != __WASI_ESUCCESS)
+        return error;
+
+    size_t to_path_len = strlen(to_path);
+
+    // Windows doesn't throw an error in the case that the new path has a
+    // trailing slash but the target to link to is a file.
+    if (to_path[to_path_len - 1] == '/'
```

Comment:
do we have a test for that in WASI?

Reply:
yes, https://github.com/WebAssembly/wasi-testsuite/blob/main/tests/rust/src/bin/path_link.rs#L139-L144